core/vm/operations_verkle.go

```diff
@@ -53,19 +53,18 @@ func gasExtCodeHash4762(evm *EVM, contract *Contract, stack *Stack, mem *Memory,
 	return evm.Accesses.TouchCodeHash(address[:], false, contract.Gas, true), nil
 }
 
-func makeCallVariantGasEIP4762(oldCalculator gasFunc) gasFunc {
+func makeCallVariantGasEIP4762(oldCalculator gasFunc, withTransferCosts bool) gasFunc {
 	return func(evm *EVM, contract *Contract, stack *Stack, mem *Memory, memorySize uint64) (uint64, error) {
 		var (
 			target           = common.Address(stack.Back(1).Bytes20())
-			transfersValue   = !stack.Back(2).IsZero()
 			witnessGas       uint64
 			_, isPrecompile  = evm.precompile(target)
 			isSystemContract = evm.isSystemContract(target)
 		)
 
 		// If value is transferred, it is charged before 1/64th
 		// is subtracted from the available gas pool.
-		if transfersValue {
+		if withTransferCosts && !stack.Back(2).IsZero() {
```

Comment:
I honestly would prefer to have the `transfersValue` to make this if more clear, but breaking up the if without the `&&` creates a "branching" complexity for the `if-elseif-else`. 

I feel safer being sure we remove L60 since not doing that adds some extra assumptions on what is on the stack for `withTransferCosts=false` situations -- so with my proposed way I access `Back(2)` only in cases that matter. In other cases that `Back(2)` would lead to the wrong semantics of the variable name.

This might feel pedantic, but I prefer not to have confusions or hacky names that only make sense in some situations.